extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcUtils.java

```diff
@@ -27,11 +29,25 @@
      * ignoring those which are located inside a pair of the double quotes.
      */
     private static final Pattern CLAIM_PATH_PATTERN = Pattern.compile("\\/(?=(?:(?:[^\"]*\"){2})*[^\"]*$)");
+    private static final Pattern JWT_PARTS_PATTERN = Pattern.compile("\\.");
 
     private OidcUtils() {
 
     }
 
+    public static boolean isOpaqueToken(String token) {
+        return JWT_PARTS_PATTERN.split(token).length != 3;
```

Comment:
Wouldn't it be a lot more efficient to just use a loop? Using a regex to count the number of '.' characters in a string feels like overkill.